### ISSUE TITLE
Dropdown does not get disabled using aria-disabled

### ISSUE THREAD

**Issue by HarisRafiq** [iTwin/iTwinUI#1685]
If we have aria-disabled set to true, the dropdown should not show the menu but it does. I have attached the link to sandbox to reproduce it.
https://codesandbox.io/s/beautiful-tess-7zzf9p?file=/src/App.tsx

**Comment by rohan-kadkol** [comment]
To disable the dropdown, I believe it would be sufficient to just disable the trigger, which in this case is the button. You can disable the `<Button>` using the `disabled` prop instead of `aria-disabled`. [Updated sandbox](https://codesandbox.io/s/quirky-haslett-5wyqf8?file=/src/App.tsx:699-742)

```diff
- <Button aria-label="More options" aria-disabled="true">
+ <Button aria-label="More options" disabled>
  Open Me
</Button
```

This is because the `aria-disabled`, whether or not to use the click handlers, and more are automatically handled by use depending on if `disabled` is `true`/`false`.

https://github.com/iTwin/iTwinUI/blob/7fd4c0ee8f15811f5b110c28da01d48564d895f4/packages/itwinui-react/src/core/utils/components/ButtonBase.tsx#L41-L48

**Comment by mayank99** [comment]
To add to this, I want to point out some misuse of ARIA attributes in the original sandbox.
- [`aria-disabled`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled) only changes semantics rather than behavior (so it doesn't disable clicks), and it is only valid on [certain roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled#associated_roles) (so you shouldn't add it to `DropdownMenu` for example).
- [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) should generally not be used if there is already text content inside the button, because it will override the accessible name and violate [WCAG SC 2.5.3](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html). (In this case the `aria-label="More options"` is the offender and is unnecessary). [Further reading](https://ashleemboyer.com/blog/don-t-overwrite-visual-labels-with-aria-label).

I would recommend against using `aria-*` attributes if you're not 100% sure how to use them. When in doubt, don't use ARIA. And make sure to run automated accessibility tests (e.g. using Lighthouse or [Axe Devtools](https://www.deque.com/axe/devtools/)) to catch invalid uses of ARIA.

**Comment by HarisRafiq** [comment]
Well thanks for the heads up. I was looking to disable DropdownMenu from v3 where the button is used from a 3rd party component which does not allow disable. In itwin-v2 DropdownMenu  had the disabled prop which made everything worked. But I solved the problem using DropdownMenu 'hidden' props.

**Comment by rohan-kadkol** [comment]
> But I solved the problem using DropdownMenu 'hidden' props.

Just to clarify, the `hidden` prop still shows the menu in the DOM

[37.webm](https://github.com/iTwin/iTwinUI/assets/45748283/732fd485-3509-45cf-8c9b-d15bf73ecc18)

**Comment by mayank99** [comment]
If you need to use it with a third-party component, you can conditionally render the DropdownMenu:
```jsx
const disabled = /* true or false */;

return disabled ? (
  <SomeButton disabled>Open me</SomeButton>
) : (
  <DropdownMenu menuItems={[…]}>
    <SomeButton>Open me</SomeButton>
  </DropdownMenu>
);
```
